packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Expressions/ValueExpressions/EnumExpression.cs

```diff
@@ -1,24 +1,18 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System;
-
 namespace Microsoft.Generator.CSharp.Expressions
 {
-    public sealed record EnumExpression(EnumType EnumType, ValueExpression Untyped) : TypedValueExpression(EnumType.Type, Untyped)
+    public sealed record EnumExpression(EnumTypeProvider EnumType, ValueExpression Untyped) : TypedValueExpression(EnumType.Type, Untyped)
     {
         public TypedValueExpression ToSerial()
-            => EnumType.SerializationMethodName is {} name
```

Comment:
we now never have a case that an enum type does not have a serialization method name, therefore i removed the other branch when this condition is `false`.